common/src/main/java/io/crate/interval/PGIntervalParser.java

```diff
@@ -99,6 +99,8 @@ static Period apply(String value) {
                         months = nullSafeIntGet(valueToken);
                     } else if (token.startsWith("day")) {
                         days = nullSafeIntGet(valueToken);
+                    } else if (token.startsWith("week")) {
+                        days = nullSafeIntGet(valueToken) * 7;
```

Comment:
Why not use a `week` variable and populate the `week` field of `Period` ?

Reply:
Weeks in psql are solely represented as days and never shown, therefore there is no need to handle them explicitly.

Reply:
```mkleen=# select interval '1 week';
 interval 
----------
 7 days
(1 row)
```